src/libs/ReportUtils.js

```diff
@@ -1185,6 +1185,26 @@ function chatIncludesChronos(report) {
                 && _.contains(report.participants, CONST.EMAIL.CHRONOS);
 }
 
+/**
+ * @param {Object} report
+ * @param {Array} sortedAndFilteredReportActions - reportActions for the report, sorted newest to oldest, and filtered for only those that should be visible
+ *
+ * @returns {String|null}
+ */
+function getNewMarkerReportActionID(report, sortedAndFilteredReportActions) {
+    if (!isUnread(report)) {
+        return null;
+    }
+
+    const newMarkerIndex = _.findLastIndex(sortedAndFilteredReportActions, reportAction => (
+        (reportAction.created || '') > report.lastReadTime
+    ));
```

Comment:
I'm not sure about this code tbh. Doesn't the dependency on having a list of `sortedAndFilteredReportActions` mean that we can only figure out _where_ to show the new marker if we have the latest report actions stored locally?

IIRC `ReportActionsView` can load with stale or no actions, but it must be initialized with the correct new marker position. In the current design the only thing required to figure out the position is a `report`. And there's a bunch of logic to update it later for various reasons - I think we also lean on the assumption that we had the right value on the first `render()` of the `ReportActionsView`.

Reply:
This might be a NAB if everyone else agrees that the whole design is kinda fundamentally borked and we'd rather do some larger refactor later. FWIW I feel like it will be very hard to solve this problem in this PR.

Reply:
Yeah I agree it could be improved but what we have here brings us up to par with what is currently on main. I think further improvements can be implemented via this convo: https://expensify.slack.com/archives/C01GTK53T8Q/p1671704053216749

Reply:
Going to resolve based on the discussion in the other comment.

Reply:
Going to resolve based on the discussion in the other comment.